src/handlers/check_commits/non_default_branch.rs

```diff
@@ -0,0 +1,44 @@
+use crate::{config::WarnNonDefaultBranchException, github::IssuesEvent};
+
+/// Returns a message if the PR is opened against the non-default branch (or the
+/// exception branch if it's an exception).
+pub(super) fn non_default_branch(
+    exceptions: &[WarnNonDefaultBranchException],
+    event: &IssuesEvent,
+) -> Option<String> {
+    let target_branch = &event.issue.base.as_ref().unwrap().git_ref;
+
+    if let Some(exception) = exceptions
+        .iter()
+        .find(|e| event.issue.title.contains(&e.title))
+    {
+        if &exception.branch != target_branch {
+            return Some(not_default_exception_branch_warn(
+                &exception.branch,
+                target_branch,
+            ));
+        }
+    } else if &event.repository.default_branch != target_branch {
+        return Some(not_default_branch_warn(
+            &event.repository.default_branch,
+            target_branch,
+        ));
+    }
+    None
+}
+
+fn not_default_branch_warn(default_: &str, target: &str) -> String {
+    format!(
+        "Pull requests are usually filed against the {default_} branch for this repo, \
+         but this one is against {target}. \
+         Please double check that you specified the right target!"
+    )
+}
+
+fn not_default_exception_branch_warn(default_: &str, target: &str) -> String {
```

Comment:
Just out of curiosity: why the underscore? `default` is not a keyword, right?

Reply:
I believe `default` is a contextual keyword, for used for specialization (in `trait` impls), but it's not the case here, I think I was confused by my editor highlighting.

Fixed.